src/components/DeFi/components/AssetInput.tsx

```diff
@@ -113,29 +113,40 @@ export const AssetInput: React.FC<AssetInputProps> = ({
   const asset = useAppSelector(state => selectAssetById(state, assetId))
 
   // Lower the decimal places when the integer is greater than 8 significant digits for better UI
-  const cryptoAmountIntegerCount = bnOrZero(bnOrZero(cryptoAmount).toFixed(0)).precision(true)
-  const formattedCryptoAmount = bnOrZero(cryptoAmountIntegerCount).isLessThanOrEqualTo(8)
-    ? cryptoAmount
-    : bnOrZero(cryptoAmount).toFixed(3)
+  const cryptoAmountIntegerCount = useMemo(
+    () => bnOrZero(bnOrZero(cryptoAmount).toFixed(0)).precision(true),
+    [cryptoAmount],
+  )
+  const formattedCryptoAmount = useMemo(
+    () =>
+      bnOrZero(cryptoAmountIntegerCount).isLessThanOrEqualTo(8)
+        ? cryptoAmount
+        : bnOrZero(cryptoAmount).toFixed(3),
+    [cryptoAmount, cryptoAmountIntegerCount],
+  )
 
   const formControlHover = useMemo(
     () => ({ bg: isReadOnly ? 'background.input.base' : 'background.input.hover' }),
     [isReadOnly],
   )
 
-  const handleValueChange = useCallback((values: NumberFormatValues) => {
-    // This fires anytime value changes including setting it on max click
-    // Store the value in a ref to send when we actually want the onChange to fire
-    amountRef.current = values.value
-  }, [])
-
```

Comment:
Moved under `handleChange` so we can consume `handleChange()` in `handleValueChange()`